File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int res = 0;
        int left = 0;
        int right = n - 1;

        int area = 0;
        while(left < right){
                area = (right - left) * Math.min(height[left], height[right]);
                res = Math.max(res, area);
                if(height[left] < height[right]) left++;
                else right--;
        }
        return res;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.c
int min(int a, int b){
  if(a<b) return a;
  else return b;
  // return a < b : a ? b;
}
int max(int a, int b){
  if(a>b) return a;
  else return b;
  // return a> b: a ? b;
}

int maxArea(int* arr, int n) {
  int left = 0;
  int right = n -1;
//   if(n == 0 ) printf("0");
  int area = 0;
  int result = 0;
  while(left<right){
    area = (right - left) * min(arr[left], arr[right]);
    result = max(area, result);

    if(arr[left] < arr[right]) left++;
    else right--;
  }
  return result;
}